File: src/main/java/com/gd/science/auth/service/FunctionService.java
package com.gd.science.auth.service;

import java.util.List;
import com.gd.science.auth.pojo.TbFunction;

import entity.PageResult;

/**
 * 服务层接口
 * 
 * @author Administrator
 *
 */
public interface FunctionService {

	/**
	 * 返回全部列表
	 * 
	 * @return
	 */
	public List<TbFunction> findAll();

	/**
	 * 返回分页列表
	 * 
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 增加
	 */
	public void add(TbFunction function);

	/**
	 * 修改
	 */
	public void update(TbFunction function);

	/**
	 * 根据ID获取实体
	 * 
	 * @param id
	 * @return
	 */
	public TbFunction findOne(Long id);

	/**
	 * 批量删除
	 * 
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * 
	 * @param pageNum  当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbFunction function, int pageNum, int pageSize);

}

File: src/main/resources/static/service/functionService.js
//服务层
app.service('functionService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../function/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../function/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../function/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../function/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../function/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../function/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../function/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});
